File: CallCenter/Models/DatXe.cs
using System;
using System.Collections.Generic;

namespace ServerService.Models;

public partial class DatXe
{
    public int DxId { get; set; }

    public DateTime DxNgayGio { get; set; }

    public string DxDiadiemdon { get; set; } = null!;

    public decimal DxGpsLon { get; set; }

    public decimal DxGpsLat { get; set; }

    public string KhTen { get; set; } = null!;

    public string KhPhone { get; set; } = null!;

    public int? KhId { get; set; }

    public int? TxId { get; set; }

    public int? XeId { get; set; }

    public int? LxId { get; set; }

    public int? TtdxId { get; set; }

}

File: ServerService/wwwroot/js/datXe.js
"use strict";

var connection = new signalR.HubConnectionBuilder().withUrl("/notificationHub").build();

connection.On < Boolean, String > ("ReceiveOrder", (check, message) => {
    console.log(message);
    if (check) {
        alert("Đã nhận đơn");
    } else {
        alert("Đơn gửi thất bại")
    }
});

connection.start().then(function () {
    console.log("Hub Connected");
}).catch(function (err) {
    return console.error(err.toString());
});
document.getElementById("send").addEventListener("click", function (event) {
    var order = {
        DxNgayGio: document.getElementById("datetime").value,
        KhTen: document.getElementById("KhTen").value,
        KhPhone: document.getElementById("KhPhone").value,
        DxDiadiemdon: document.getElementById("address").value,
        LxId: document.getElementById("Lxid").value,
        DxGpsLat: document.getElementById("latitude").value,
        DxGpsLon: document.getElementById("longitude").value,
    }
    var message = JSON.stringify(order);
    // Gửi tin nhắn đến SignalR Hub
    connection.invoke("SendOrder", message).catch(function (err) {
        return console.error(err.toString());
    });
    event.preventDefault();
});
